test/basic/vpc_update/vpc_update_test.go

```diff
@@ -0,0 +1,123 @@
+package vpc_update
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	legacy_aws_sdk_ec2 "github.com/aws/aws-sdk-go/service/ec2"
+	terratest_aws "github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+
+	cac "github.com/aptible/cloud-api-clients/clients/go"
+	"github.com/aptible/terraform-provider-aptible-iaas/internal/client"
+)
+
+var mutableTFVariables = map[string]interface{}{
+	"organization_id": os.Getenv("ORGANIZATION_ID"),
+	"environment_id":  os.Getenv("ENVIRONMENT_ID"),
+	"aptible_host":    os.Getenv("APTIBLE_HOST"),
+	"vpc_name":        "test-vpc",
+}
+
+func assertCommonValues(t *testing.T, vpcId string, vpcAsset *cac.AssetOutput, vpcAws []*terratest_aws.Vpc) {
+	assert.Equal(t, vpcAsset.Id, vpcId)
+	assert.Equal(t, vpcAsset.Status, cac.ASSETSTATUS_DEPLOYED)
+	assert.GreaterOrEqual(t, len(vpcAws), 1)
+	assert.Equal(t, len(vpcAws[0].Subnets), 6)
+}
+
+// generateMutableTerraformOptions - Generates a new pointer and object of mutable reference of the variable map,
+// which is mutated over the course of the test suite to avoid specifying the full set
+func generateMutableTerraformOptions() *terraform.Options {
+	return &terraform.Options{
+		TerraformDir: ".",
+		Vars:         mutableTFVariables,
+	}
+}
+
+func reachabilityAnalysisAssertions(t *testing.T) {
+	privInstanceENI := terraform.Output(t, generateMutableTerraformOptions(), "test_instance_private_eni")
```

Comment:
why are you passing in new pointer each time?

Reply:
it's the same pointer, so the same testing T thing

Reply:
OHHH you mean the top level config one? because i think it expects one, i can just generate one and re-ref, didn't consider it important as the map values is a pointer too for its vals

Reply:
laziness/copy pasta - will ref one val in this func